### PR TITLE
Fix #8020: Add missing docking tiles around industry neutral stations

### LINKED ISSUE #8020: Not all tiles around an oil rig are docking tiles
## Version of OpenTTD
1.10.0-RC1
## Expected result
Oil Rig Industry and Station tiles both have docking tiles around.
## Actual result
Oil Rig Station tile doesn't have docking tiles around itself.
## Steps to reproduce
Tuningville Transport, 1973-09-08

### PR DESCRIPTION
**Pull request by SamuXarick**

### CODE REVIEW

**Comment by LordAro** [comment]
I'm not seeing why a savegame bump is necessary here

**Comment by glx22** [comment]
It's to fix "broken" saves since SLV_MULTITILE_DOCKS

File: src/saveload/afterload.cpp [hunk 1]
```diff
@@ -3123,6 +3123,12 @@ bool AfterLoadGame()
 		}
 	}
 
+	if (IsSavegameVersionBefore(SLV_FIX_NEUTRAL_STATION_DOCKING_TILE)) {
```
Comment (glx22) [inline src/saveload/afterload.cpp]:
```suggestion
	if (IsSavegameVersionBefore(SLV_FIX_NEUTRAL_STATION_DOCKING_TILE) && !IsSavegameVersionBefore(SLV_MULTITILE_DOCKS)) {
```
to not iterate all stations twice, as now the original `IsSavegameVersionBefore(SLV_MULTITILE_DOCKS)` block above this fix will be correct.

Comment (glx22) [reply src/saveload/afterload.cpp]: Other option is to extract the iteration from `IsSavegameVersionBefore(SLV_MULTITILE_DOCKS)` block and run it unconditionally like `AfterLoadRoadStops()`, removing the need for the savegame bump

**Comment by LordAro** [comment]
I'm prepared to say that the few missing tiles don't matter for savegames created between SLV_MULTIDOCK_TILES and now. There's not been a (stable) release between then anyway

**Comment by nielsmh** [comment]
Agree with @LordAro here, a bump is not strictly necessary. The missing tiles mean some water industries won't have ships docking in all positions but it's not a game-breaking catastrophe.
On the contrary, a savegame bump will prevent the fix from being backported to 1.10 release, since the savegame version should be frozen in release branches.

**Review by LordAro — APPROVED** [review]

**Pull request merged**